### PR TITLE
 CON-8 Abass/add category

### PR DESCRIPTION
**Pull request by abassawo**
As a contest creator, I want to add a scoring category to my contest, specifying name and range (stretch goal).

### CODE REVIEW

File: app/src/main/java/io/intrepid/contest/models/Category.java [hunk 1]
```diff
@@ -9,6 +11,17 @@ public Category(String name, String description) {
         this.description = description;
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        try {
+            Category otherCategory = (Category) obj;
+            return this.name == otherCategory.name &&
```
Comment (jmccormack200) [inline app/src/main/java/io/intrepid/contest/models/Category.java]: are you sure you want == and not .equals?

Comment (abassawo) [inline app/src/main/java/io/intrepid/contest/models/Category.java]: Good catch!

File: app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java [hunk 2]
```diff
@@ -84,7 +85,13 @@ public void addCategory(Category category) {
 
     @Override
     public void onNewCategoryAdded(Intent data) {
-        //todo - Add category
+        String categoryName = data.getStringExtra("NAME");
+        String categoryDescription = data.getStringExtra("DESCRIPTION");
```
Comment (gsintrepid) [inline app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java]: You have these as constants in `AddCategoryActivity`, maybe you can reuse them in this file. @jmccormack200 is it ok to make them constants in the presenter and have the activity use them?

File: app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java [hunk 3]
```diff
@@ -40,7 +40,8 @@ public void onBackButtonClicked() {
         int currentIndex = view.getCurrentIndex();
         if (currentIndex == 0) {
             view.cancelEdit();
-        } else {
+        }
+        else {
```
Comment (gsintrepid) [inline app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java]: It seems like the file was unformatted?

File: app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java (hunk 2 above)
Comment (jmccormack200) [inline app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java]: Actually this is a good catch, Intent is an android type object so we dont want to move the whole intent into the presenter, we would parse the data out in Activity and then pass the parsed data into the presenter.

Comment (abassawo) [inline app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java]: Yeah I agree. I actually revisited this and changed it in a later branch, but forgot to go back here and update too.

**Pull request merged**